### DISCUSSION TITLE
lesson 14 error

### DISCUSSION THREAD

**Discussion by GenevaRoc** [Cyfrin/foundry-full-course-cu#1224] [General]
I keep getting this error in my setUp trying to run a test. 
Here is the error 
`[FAIL. Reason: Setup failed: 1e4fbdf7:] setUp() (gas: 0)
Traces:
  [7051917] MyGovernorTest::setUp() 
    ├─ [1611226] → new MyToken@0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f
    │   └─ ← 7816 bytes of code
    ├─ [97013] MyToken::mint(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], 100000000000000000000 [1e20]) 
    │   ├─ emit Transfer(from: 0x0000000000000000000000000000000000000000, to: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], value: 100000000000000000000 [1e20])
    │   └─ ← ()
    ├─ [26993] MyToken::delegate(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D]) 
    │   ├─ emit DelegateChanged(delegator: MyGovernorTest: [0x7FA9385bE102ac3EAc297483Dd6233D62b3e1496], fromDelegate: 0x0000000000000000000000000000000000000000, toDelegate: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   └─ ← ()
    ├─ [0] VM::startPrank(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D]) 
    │   └─ ← ()
    ├─ [70257] MyToken::delegate(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D]) 
    │   ├─ emit DelegateChanged(delegator: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], fromDelegate: 0x0000000000000000000000000000000000000000, toDelegate: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   ├─ emit DelegateVotesChanged(delegate: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], previousVotes: 0, newVotes: 100000000000000000000 [1e20])
    │   └─ ← ()
    ├─ [1425854] → new TimeLock@0x7BD1119CEC127eeCDBa5DCA7d1Bd59986f6d7353
    │   ├─ emit RoleGranted(role: 0x0000000000000000000000000000000000000000000000000000000000000000, account: TimeLock: [0x7BD1119CEC127eeCDBa5DCA7d1Bd59986f6d7353], sender: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   ├─ emit RoleGranted(role: 0x0000000000000000000000000000000000000000000000000000000000000000, account: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], sender: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   ├─ emit MinDelayChange(oldDuration: 0, newDuration: 3600)
    │   └─ ← 6754 bytes of code
    ├─ [3560321] → new MyGovernor@0x5929B14F2984bBE5309c2eC9E7819060C31c970f
    │   ├─ emit VotingDelaySet(oldVotingDelay: 0, newVotingDelay: 7200)
    │   ├─ emit VotingPeriodSet(oldVotingPeriod: 0, newVotingPeriod: 50400 [5.04e4])
    │   ├─ emit ProposalThresholdSet(oldProposalThreshold: 0, newProposalThreshold: 0)
    │   ├─ [427] MyToken::clock() [staticcall]
    │   │   └─ ← 1
    │   ├─ emit QuorumNumeratorUpdated(oldQuorumNumerator: 0, newQuorumNumerator: 4)
    │   ├─ emit TimelockChange(oldTimelock: 0x0000000000000000000000000000000000000000, newTimelock: TimeLock: [0x7BD1119CEC127eeCDBa5DCA7d1Bd59986f6d7353])
    │   └─ ← 17162 bytes of code
    ├─ [285] TimeLock::PROPOSER_ROLE() [staticcall]
    │   └─ ← 0xb09aa5aeb3702cfd50b6b62bc4532604938f21248a27a1d5ca736082b6819cc1
    ├─ [286] TimeLock::EXECUTOR_ROLE() [staticcall]
    │   └─ ← 0xd8aa0f3194971a2a116679f7c2090f6939c8d4e01a2a8d7e41d55e5351469e63
    ├─ [262] TimeLock::DEFAULT_ADMIN_ROLE() [staticcall]
    │   └─ ← 0x0000000000000000000000000000000000000000000000000000000000000000
    ├─ [27571] TimeLock::grantRole(0xb09aa5aeb3702cfd50b6b62bc4532604938f21248a27a1d5ca736082b6819cc1, MyGovernor: [0x5929B14F2984bBE5309c2eC9E7819060C31c970f]) 
    │   ├─ emit RoleGranted(role: 0xb09aa5aeb3702cfd50b6b62bc4532604938f21248a27a1d5ca736082b6819cc1, account: MyGovernor: [0x5929B14F2984bBE5309c2eC9E7819060C31c970f], sender: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   └─ ← ()
    ├─ [27571] TimeLock::grantRole(0xd8aa0f3194971a2a116679f7c2090f6939c8d4e01a2a8d7e41d55e5351469e63, 0x0000000000000000000000000000000000000000) 
    │   ├─ emit RoleGranted(role: 0xd8aa0f3194971a2a116679f7c2090f6939c8d4e01a2a8d7e41d55e5351469e63, account: 0x0000000000000000000000000000000000000000, sender: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   └─ ← ()
    ├─ [4468] TimeLock::revokeRole(0x0000000000000000000000000000000000000000000000000000000000000000, user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D]) 
    │   ├─ emit RoleRevoked(role: 0x0000000000000000000000000000000000000000000000000000000000000000, account: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D], sender: user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   └─ ← ()
    ├─ [0] VM::stopPrank() 
    │   └─ ← ()
    ├─ [319] → new <Unknown>@0x2e234DAe75C793f67A35089C9d99245E1C58470b
    │   └─ ← 0 bytes of code
    └─ ← "1e4fbdf7:"`

And here is my code base 
`contract MyGovernorTest is Test {
    MyGovernor governor;
    Box box;
    TimeLock timeLock;
    MyToken myToken;

    address public USER = makeAddr("user");
    uint256 public constant INITIAL_SUPPLY = 100 ether;

    address[] proposers;
    address[] executors;
    //address admin;
    address public initialOwner;

    uint256 public constant MIN_DELAY = 3600;

    function setUp() public {
        myToken = new MyToken();
        myToken.mint(USER, INITIAL_SUPPLY);
        myToken.delegate(USER);

        vm.startPrank(USER);
        myToken.delegate(USER);
        timeLock = new TimeLock(MIN_DELAY, proposers, executors); //././
        governor = new MyGovernor(myToken, timeLock);

        bytes32 proposerRole = timeLock.PROPOSER_ROLE();
        bytes32 executorRole = timeLock.EXECUTOR_ROLE();
        bytes32 adminRole = timeLock.DEFAULT_ADMIN_ROLE();

        timeLock.grantRole(proposerRole, address(governor));
        timeLock.grantRole(executorRole, address(0));
        timeLock.revokeRole(adminRole, USER);
        vm.stopPrank();

        box = new Box(initialOwner);
        //console.log("Box :", address(box));
        box.transferOwnership(address(timeLock));
    }

    function testCantUpdateBoxWithOutGovernance() public {
        vm.expectRevert();
        box.store(1);
    }
}`

**Comment by GenevaRoc** [comment]
@theirrationalone

**Comment by Bogomil-Simitchiev** [comment]
I have the same issue -> [FAIL. Reason: Setup failed: 1e4fbdf7:] setUp() 
Did you fix that?
